internal/service/outposts/outpost_assets_data_source.go

```diff
@@ -25,6 +28,31 @@ func DataSourceOutpostAssets() *schema.Resource {
 				Computed: true,
 				Elem:     &schema.Schema{Type: schema.TypeString},
 			},
+			"host_id_filter": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.All(
+						validation.StringLenBetween(1, 50),
+						validation.StringMatch(regexp.MustCompile(`^^[A-Za-z0-9-]*$`), "must match [a-zA-Z0-9-]"),
+					),
+				},
+			},
+			"status_id_filter": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 2,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.All(
+						validation.StringInSlice([]string{
+							"ACTIVE",
+							"RETIRING",
+						}, false),
```

Comment:
We prefer using constants defined in the SDK vs hardcoded strings - the SDK has a function to return all of the enum values: https://docs.aws.amazon.com/sdk-for-go/api/service/outposts/#AssetState_Values

```suggestion
					ValidateFunc: validation.All(
						validation.StringInSlice(AssetState_Values(), false),
```